tensorflow/python/keras/engine/compile_utils_test.py

```diff
@@ -356,6 +359,34 @@ def __call__(self, y_true, y_pred):
     self.assertEqual(loss_container._losses[0].name, 'custom_loss_fn')
     self.assertEqual(loss_container._losses[1].name, 'custom_loss_class')
 
+  def test_ragged_tensor_output(self):
+    """ Ensure that ragged tensors can be passed as targets and predictions.
+    """
+    def custom_loss_fn(y_true, y_pred):
+      losses = ragged_functional_ops.map_flat_values(
+        losses_mod.mse, y_true, y_pred)
+      return math_ops.reduce_mean(losses)
+
+    class CustomLossClass(object):
+
+      def __call__(self, y_true, y_pred):
+        losses = ragged_functional_ops.map_flat_values(
+          losses_mod.mse, y_true, y_pred)
+        return math_ops.reduce_mean(losses)
+
+    loss_container = compile_utils.LossesContainer(
+        [custom_loss_fn, CustomLossClass()])
+
+    v_t = constant_op.constant([[3., 4.], [1., 2.], [3., 5.]])
+    v_p = constant_op.constant([[3.1, 4.], [1., 2.], [3., 5.]])
+
+    y_t = ragged_array_ops.expand_dims(
```

Comment:
You can just use `tf.expand_dims(ragged_tensor.RaggedTensor.from_row_splits(...))` -- that will automatically delegate to ragged_array_ops.expand_dims if its argument is ragged.

Reply:
Done.